data_steward/validation/main.py

```diff
@@ -95,33 +101,35 @@ def run_export(datasource_id=None, folder_prefix="", target_bucket=None):
     :param target_bucket: Bucket to save report. If None, use bucket associated with hpo_id.
     """
     results = []
-
+    project_id = app_identity.get_application_id()
+    storage_client = StorageClient(project_id)
     # Using separate var rather than hpo_id here because hpo_id None needed in calls below
     if datasource_id is None and target_bucket is None:
         raise RuntimeError(
             f"Cannot export if neither hpo_id nor target_bucket is specified.")
     else:
-        datasource_name = datasource_id
         if target_bucket is None:
-            target_bucket = gcs_utils.get_hpo_bucket(datasource_id)
+            target_bucket: Bucket = storage_client.get_hpo_bucket(datasource_id)
+        else:
+            target_bucket: Bucket = storage_client.bucket(target_bucket)
 
     logging.info(
-        f"Exporting {datasource_name} report to bucket {target_bucket}")
+        f"Exporting {datasource_id} report to bucket {target_bucket.name}")
 
     # Run export queries and store json payloads in specified folder in the target bucket
-    reports_prefix = folder_prefix + ACHILLES_EXPORT_PREFIX_STRING + datasource_name + '/'
+    reports_prefix: str = f'{folder_prefix}{ACHILLES_EXPORT_PREFIX_STRING}{datasource_id}/'
     for export_name in common.ALL_REPORTS:
         sql_path = os.path.join(export.EXPORT_PATH, export_name)
         result = export.export_from_path(sql_path, datasource_id)
         content = json.dumps(result)
         fp = StringIO(content)
-        result = gcs_utils.upload_object(target_bucket,
-                                         reports_prefix + export_name + '.json',
-                                         fp)
+        blob: Blob = target_bucket.blob(f'{reports_prefix}{export_name}.json')
+        blob.upload_from_file(fp)
+        result: dict = storage_client.get_blob_metadata(blob)
         results.append(result)
     result = save_datasources_json(datasource_id=datasource_id,
                                    folder_prefix=folder_prefix,
-                                   target_bucket=target_bucket)
+                                   target_bucket=target_bucket.name)
```

Comment:
Can you pass in the storage client as well to it can be reused here?

Reply:
I should've mentioned it should be a required parameter. All such functions which are invoked with an available storage client should use it without having to create it again or check for its availability.
```suggestion
    result = save_datasources_json(storage_client=storage_client,
                                   datasource_id=datasource_id,
                                   folder_prefix=folder_prefix,
                                   target_bucket=target_bucket.name)
```